### DISCUSSION TITLE
Exceeding Sotrage Limits?

### DISCUSSION THREAD

**Discussion by Carlos-err406** [supabase/supabase#31056] [Questions] [answered]
Hello Supabase Team. 

First of all i would like to say I love what you have accomplished with this product. 

For a couple of weeks now I've been getting a warning that my organization is exceding storage usage limits (see attachments) 

![Screenshot 2024-12-10 110957](https://github.com/user-attachments/assets/f682418e-d879-4e2f-9cfe-74697f6d3f88)

![Screenshot 2024-12-10 111315](https://github.com/user-attachments/assets/163fbfcc-cc8f-4a33-a776-64309be9681f)

But i make sure to periodically delete the files from the storage. On Friday, December 6 I even deleted most of the storage objects, from 401 objects to 170, down to 0.04GB (summing up the size from the storage.objects table of the project that had the most objects by far) But that was not reflected on the Storage Usage Chart (see attachment) 

![Screenshot 2024-12-10 112030](https://github.com/user-attachments/assets/f7eae1f3-6f23-4630-9b37-fc52b59de269)

Of course i deleted from storage.objects in the sql editor (deleted all objects older than 2 weeks). And made sure they were deleted from the bucket. 

So my main concern here is why i keep getting this warning despite i deleted over a half the items of my storage. 

Greetings, Carlos Daniel Vilaseca

**Comment by GaryAustin1** [comment, marked as answer]
You cannot delete files by deleting the storage.objects table rows. You have to use UI or strorage API calls.  Any files you deleted using SQL on storage.objects are now orphaned in S3. I personally don’t know of a way to delete them now.

**Reply by Carlos-err406** [reply]
is there a way to backup the current bucket so i can delete de proyect and create it again?

**Reply by GaryAustin1** [reply]
This is the only method I’ve seen. https://supabase.com/docs/guides/platform/migrating-within-supabase#migrate-storage-objects

**Reply by Carlos-err406** [reply]
i tried a similar approach, it worked fine, managed to backup locally all the data and storage objects

i just hope that deleting the supabase project also deletes the orphan objects of the S3

**Reply by GaryAustin1** [reply]
If you create a new instance it will have its own S3 partition.

**Reply by afex-git** [reply]
The only thing that helped me was "Transfer project" to another organization (it can be done through the UI in just a couple of clicks). All APIs are preserved, and there’s no need to change anything in the env. It took about 20 minutes, and the transfer was done.

**Reply by nguyenthanh1205tb** [reply]
> The only thing that helped me was "Transfer project" to another organization (it can be done through the UI in just a couple of clicks). All APIs are preserved, and there’s no need to change anything in the env. It took about 20 minutes, and the transfer was done.

I've never tried this before. Will it preserve the database? And won't transferring the project like this also transfer orphaned files?

**Reply by GaryAustin1** [reply]
@nguyenthanh1205tb Support has a way to delete your orphaned data.
